eslint.config.mjs

```diff
@@ -0,0 +1,1393 @@
+						'vs/base/*/~',
+						'vs/base/parts/*/~',
+						'vs/platform/*/~',
+						'vs/editor/~',
+						'vs/editor/contrib/*/~',
+						'vs/editor/editor.all.js',
+						'vs/workbench/~',
+						'vs/workbench/api/~',
+						'vs/workbench/services/*/~',
+						'vs/workbench/contrib/*/~',
+						'vs/workbench/workbench.common.main.js'
+					]
+				},
+				{
+					'target': 'src/vs/amdX.ts',
+					'restrictions': [
+						'vs/base/common/*'
+					]
+				},
+				{
+					'target': 'src/vs/{loader.d.ts,monaco.d.ts,nls.ts,nls.messages.ts}',
+					'restrictions': []
+				},
+				{
+					'target': 'src/vscode-dts/**',
+					'restrictions': []
+				},
+				{
+					'target': 'src/bootstrap-window.ts',
+					'restrictions': []
+				},
+				{
+					'target': 'src/vs/nls.ts',
+					'restrictions': [
+						'vs/*'
+					]
+				},
+				{
+					'target': 'src/{bootstrap-cli.ts,bootstrap-esm.ts,bootstrap-fork.ts,bootstrap-import.ts,bootstrap-meta.ts,bootstrap-node.ts,bootstrap-server.ts,cli.ts,main.ts,server-cli.ts,server-main.ts}',
+					'restrictions': [
+						'vs/**/common/*',
+						'vs/**/node/*',
+						'vs/nls.js',
+						'src/*.js',
+						'*' // node.js
+					]
+				}
+			]
+		}
+	},
+	{
+		files: [
+			'test/**/*.ts'
+		],
+		languageOptions: {
+			parser: tseslint.parser,
+		},
+		plugins: {
+			'local': pluginLocal,
+		},
+		rules: {
+			'local/code-import-patterns': [
+				'warn',
+				{
+					'target': 'test/smoke/**',
+					'restrictions': [
+						'test/automation',
+						'test/smoke/**',
+						'@vscode/*',
+						'@parcel/*',
+						'@playwright/*',
+						'*' // node modules
+					]
+				},
+				{
+					'target': 'test/automation/**',
+					'restrictions': [
+						'test/automation/**',
+						'@vscode/*',
+						'@parcel/*',
+						'playwright-core/**',
+						'@playwright/*',
+						'*' // node modules
+					]
+				},
+				{
+					'target': 'test/integration/**',
+					'restrictions': [
+						'test/integration/**',
+						'@vscode/*',
+						'@parcel/*',
+						'@playwright/*',
+						'*' // node modules
+					]
+				},
+				{
+					'target': 'test/monaco/**',
+					'restrictions': [
+						'test/monaco/**',
+						'@vscode/*',
+						'@parcel/*',
+						'@playwright/*',
+						'*' // node modules
+					]
+				}
+			]
+		}
+	},
+	{
+		files: [
+			'src/vs/workbench/contrib/notebook/browser/view/renderers/*.ts'
+		],
+		languageOptions: {
+			parser: tseslint.parser,
+		},
+		plugins: {
+			'local': pluginLocal,
+		},
+		rules: {
+			'local/code-no-runtime-import': [
+				'error',
+				{
+					'src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts': [
+						'**/*'
+					]
+				}
+			],
+			'local/code-limited-top-functions': [
+				'error',
+				{
+					'src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts': [
+						'webviewPreloads',
+						'preloadsScriptStr'
+					]
+				}
+			]
+		}
+	},
+	// Terminal
+	{
+		files: [
+			'src/vs/workbench/contrib/terminal/**/*.ts'
+		],
```

Comment:
Can you add `'src/vs/workbench/contrib/terminalContrib/**/*.ts'` here too? You'll also need to delete it at https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/terminalContrib/.eslintrc.json

Reply:
Ah sorry, yes missed that we had some `eslintrc.json` too. Will make sure those also get ported

Reply:
@Tyriar why do we have component specific eslint rules? Isn't the whole idea that we have team agreement and can feel at home anywhere?

Reply:
DMed you